src/app/core/admin/admin-entity/admin-entity-general-settings/conditional-color-config/conditions-editor/conditions-editor.component.ts

```diff
@@ -0,0 +1,183 @@
+import {
+  Component,
+  Input,
+  Output,
+  EventEmitter,
+  OnInit,
+  inject,
+} from "@angular/core";
+import { MatButtonModule } from "@angular/material/button";
+import { MatFormFieldModule } from "@angular/material/form-field";
+import { MatSelectModule } from "@angular/material/select";
+import { MatTooltipModule } from "@angular/material/tooltip";
+import { FontAwesomeModule } from "@fortawesome/angular-fontawesome";
+import { FormControl, ReactiveFormsModule } from "@angular/forms";
+import { EntityConstructor } from "app/core/entity/model/entity";
+import { SimpleDropdownValue } from "app/core/common-components/basic-autocomplete/simple-dropdown-value.interface";
+import { FormFieldConfig } from "app/core/common-components/entity-form/FormConfig";
+import { EntitySchemaService } from "app/core/entity/schema/entity-schema.service";
+import { DynamicEditComponent } from "app/core/entity/entity-field-edit/dynamic-edit/dynamic-edit.component";
+import { MatDialog } from "@angular/material/dialog";
+import { JsonEditorDialogComponent } from "app/core/admin/json-editor/json-editor-dialog/json-editor-dialog.component";
+
+/**
+ * Reusable component for editing conditions (field-value pairs) with JSON support
+ */
+@Component({
+  selector: "app-conditions-editor",
+  templateUrl: "./conditions-editor.component.html",
+  styleUrls: ["./conditions-editor.component.scss"],
+  imports: [
+    MatButtonModule,
+    MatFormFieldModule,
+    MatSelectModule,
+    MatTooltipModule,
+    FontAwesomeModule,
+    DynamicEditComponent,
+    ReactiveFormsModule,
+  ],
+})
+export class ConditionsEditorComponent implements OnInit {
+  @Input() conditions: any = { $or: [] };
+  @Input() entityConstructor: EntityConstructor;
+  @Input() fieldOptions: SimpleDropdownValue[] = [];
+
+  @Output() conditionsChange = new EventEmitter<any>();
+
+  conditionFormFieldConfigs = new Map<string, FormFieldConfig>();
+  conditionFormControls = new Map<string, FormControl>();
+
+  private readonly entitySchemaService = inject(EntitySchemaService);
+  private readonly dialog = inject(MatDialog);
+
+  ngOnInit(): void {
+    if (!this.entityConstructor) return;
+    this.rebuildFormConfigs();
+  }
+
+  /**
+   * Get the conditions array
+   */
+  getConditionsArray(): any[] {
+    return this.conditions?.$or || [];
```

Comment:
can we access this directly in the template? Or store it as a signal or something? Using methods recalculates things a lot during change detection and is not good for performance in Angular templates